examples/ShoppingAds/AddShoppingSmartAd.php

```diff
@@ -0,0 +1,424 @@
+     * @param int $customerId the client customer ID
+     * @param string $campaignResourceName the resource name of the campaign that
+     *     the new ad group will belong to
+     * @return string the resource name of the newly created ad group
+     */
+    private static function addSmartShoppingAdGroup(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        string $campaignResourceName
+    ) {
+        // Creates an ad group.
+        $adGroup = new AdGroup([
+            'name' => new StringValue(['value' => 'Earth to Mars Cruise #' . uniqid()]),
+            'campaign' => new StringValue(['value' => $campaignResourceName]),
+            // Sets the ad group type to SHOPPING_SMART_ADS. This cannot be set to other types.
+            'type' => AdGroupType::SHOPPING_SMART_ADS,
+            'status' => AdGroupStatus::ENABLED
+        ]);
+
+        // Creates an ad group operation.
+        $adGroupOperation = new AdGroupOperation();
+        $adGroupOperation->setCreate($adGroup);
+
+        // Issues a mutate request to add the ad group.
+        $adGroupServiceClient = $googleAdsClient->getAdGroupServiceClient();
+        $response = $adGroupServiceClient->mutateAdGroups($customerId, [$adGroupOperation]);
+
+        /** @var AdGroup $addedAdGroup */
+        $addedAdGroup = $response->getResults()[0];
+        printf(
+            "Added a Smart Shopping ad group with resource name: '%s'.%s",
+            $addedAdGroup->getResourceName(),
+            PHP_EOL
+        );
+
+        return $addedAdGroup->getResourceName();
+    }
+
+    /**
+     * Creates a new ad group ad in the specified Smart Shopping ad group.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the client customer ID
+     * @param string $adGroupResourceName the resource name of the ad group that
+     *     the new ad group ad will belong to
+     */
+    private static function addSmartShoppingAdGroupAd(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        string $adGroupResourceName
+    ) {
+        // Creates a new ad group ad.
+        $adGroupAd = new AdGroupAd([
+            // Sets a new Smart Shopping ad.
+            'ad' => new Ad(['shopping_smart_ad' => new ShoppingSmartAdInfo()]),
+            // Sets the ad group.
+            'ad_group' => new StringValue(['value' => $adGroupResourceName])
+        ]);
+
+        // Creates an ad group ad operation.
+        $adGroupAdOperation = new AdGroupAdOperation();
+        $adGroupAdOperation->setCreate($adGroupAd);
+
+        // Issues a mutate request to add the ad group ad.
+        $adGroupAdServiceClient = $googleAdsClient->getAdGroupAdServiceClient();
+        $response = $adGroupAdServiceClient->mutateAdGroupAds($customerId, [$adGroupAdOperation]);
+
+        /** @var AdGroupAd $addedAdGroupAd */
+        $addedAdGroupAd = $response->getResults()[0];
```

Comment:
Same comment as above, although you are calling getResourceName() just once here.

Reply:
Yes only one call so I think it is good as it is, right? I could do it inline but I think it is better to clarify the type returned like it is done in the other functions.

Reply:
SGTM.